playwright/_impl/_transport.py

```diff
@@ -79,21 +121,73 @@ async def run(self) -> None:
                         buffer = buffer + data
                     else:
                         buffer = data
-                obj = json.loads(buffer)
 
-                if "DEBUGP" in os.environ:  # pragma: no cover
-                    print("\x1b[33mRECV>\x1b[0m", json.dumps(obj, indent=2))
+                obj = self.deserialize_message(buffer)
                 self.on_message(obj)
             except asyncio.IncompleteReadError:
                 break
             await asyncio.sleep(0)
         self._stopped_future.set_result(None)
 
     def send(self, message: Dict) -> None:
-        msg = json.dumps(message)
-        if "DEBUGP" in os.environ:  # pragma: no cover
-            print("\x1b[32mSEND>\x1b[0m", json.dumps(message, indent=2))
-        data = msg.encode()
+        data = self.serialize_message(message)
         self._output.write(
             len(data).to_bytes(4, byteorder="little", signed=False) + data
         )
+
+
+class WebSocketTransport(AsyncIOEventEmitter, Transport):
+    def __init__(self, ws_endpoint: str, timeout: float) -> None:
+        super().__init__()
+        Transport.__init__(self)
+
+        self._stopped = False
+        self.ws_endpoint = ws_endpoint
+        self.timeout = timeout
+        self._loop: asyncio.AbstractEventLoop
+
+    def stop(self) -> None:
+        self._stopped = True
+        self._loop.create_task(self._connection.close())
```

Comment:
Declare self._connection in __init__, did our mypy break?

Reply:
Yes its broken currently, will follow up to fix it.

Reply:
It can automatically infer that I set it in WebSocketTransport.run() 

But yes your assumption was also correct that mypy was broken. Fixed in #643